aws/resource_aws_api_gateway_stage.go

```diff
@@ -20,6 +20,20 @@ func resourceAwsApiGatewayStage() *schema.Resource {
 		Read:   resourceAwsApiGatewayStageRead,
 		Update: resourceAwsApiGatewayStageUpdate,
 		Delete: resourceAwsApiGatewayStageDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected REST-API-ID/STAGE-NAME", d.Id())
+				}
+				restApiID := idParts[0]
+				stageName := idParts[1]
+				d.Set("stage_name", stageName)
+				d.Set("rest_api_id", restApiID)
+				d.SetId(fmt.Sprintf("ags-%s-%s", restApiID, stageName))
```

Comment:
(as in the other PR's) - why not set the ID and then call read?

Reply:
Noting for future posterity: in these PRs we're just doing the minimum to support resource import (with a consistent format of `REST-API-ID/THING`) -- we can eventually switch to read/update/delete via the resource ID, but this implementation provides the bare minimum needed for import without the unrelated code changes right now.